tests/test_pages.py

```diff
@@ -0,0 +1,98 @@
+# pylint: disable=redefined-outer-name
+from __future__ import print_function
+
+from collections import namedtuple
+
+import pytest
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+User = namedtuple('User', ['username', 'password', 'name'])
+
+PAGES = [
```

Comment:
what's an easy way to extend that list for plugins?

Reply:
Longer term I would like a REST endpoint that gets all pages visible to the user. That way the test basically becomes "verify the menu". If the endpoint returns a 404 for compatibility, it'd fall back to this list. If it's empty, it would raise an error.

Reply:
Makes sense for a quickly moving upstream.

Might need some tuning downstream where we want to test also older versions with plugins. But this can be added later (or worst case by having a fork for downstream)

Reply:
We can move it to a fallback file where every line is a URL. In downstream we'd just make sure fallback file has all the paths we want to check.

Reply:
That sounds good, yeah.

Reply:
You could for now at least move this out to a folder to read files from. Then plugin authors can at least add a list of their pages via PRs to be included in the test. If the path provided 404s, then assume that plugin is not currently installed when running the test.

Reply:
https://github.com/theforeman/foreman/pull/6966 is the PR for the menu endpoint. I like the idea of a urls.d directory for fallback.